src/thunderbots/software/util/parameter/parameter.h

```diff
@@ -71,11 +72,50 @@ class Parameter
         if (Parameter<T>::getMutableRegistry().count(this->name_))
         {
             auto& param_in_registry = Parameter<T>::getMutableRegistry().at(this->name_);
-            return param_in_registry->value_;
+
+            param_in_registry.first->lock();
+            auto value = param_in_registry.second->value_;
+            param_in_registry.first->unlock();
+
+            return value;
+        }
+
+        // TODO fix this on ROS removal to throw an exception, ideally we do not
+        // want to return a value here as params that aren't in the registry are not
+        // threadsafe
```

Comment:
Could we associate this with a ticket just in case we forget?

Reply:
added a ticket and updated comment